File: IQ/Classes/clsScreenOverride.vb
Imports System.Runtime.Serialization
Imports dataAccess

<DataContract>
Public Class clsScreenOverride
    <DataMember>
    Public AccountID As Integer
    <DataMember>
    Public ScreenID As Integer
    <DataMember>
    Public Path As String
    <DataMember>
    Public FieldId As Integer
    <DataMember>
    Public ForceVisibilityTo As Nullable(Of Boolean)
    <DataMember>
    Public ForceOrderTo As Nullable(Of Integer)
    <DataMember>
    Public ForceWidthTo As Nullable(Of Double)
    <DataMember>
    Public ForceSortTo As String
    <DataMember>
    Public ForceFilterTo As String
    <DataMember>
    Public FieldName As String
    <DataMember>
    Public DisplayUnit As clsUnit

  
    Public Sub New(AccountID As Integer, ScreenID As Integer, BranchPath As String, FieldId As Integer, ForceVisibilityTo As Nullable(Of Boolean), ForceOrderTo As Nullable(Of Integer), ForceWidthTo As Nullable(Of Double), ForceSortTo As String, ForceFilterTo As String,DisplayUnit As clsUnit)
        Me.AccountID = AccountID
        Me.FieldId = FieldId
        Me.ForceVisibilityTo = ForceVisibilityTo
        Me.ForceOrderTo = ForceOrderTo
        Me.ScreenID = ScreenID
        Me.Path = BranchPath
        Me.ForceWidthTo = ForceWidthTo
        Me.ForceSortTo = ForceSortTo
        Me.ForceFilterTo = ForceFilterTo
        Me.DisplayUnit = DisplayUnit
        iq.ScreenOverrides.Add(Me)
    End Sub

    Public Sub New()

    End Sub

    Public Function Update() As Boolean

        Dim sql$
        sql$ = "Update AccountScreenOverride set ForceVisibilityTo = " & If(Me.ForceVisibilityTo Is Nothing, "NULL", da.SqlEncode(Me.ForceVisibilityTo)) & ",ForceOrderTo = " & If(Me.ForceOrderTo Is Nothing, "NULL", da.SqlEncode(Me.ForceOrderTo)) & ",ForceWidthTo = " & If(Me.ForceWidthTo Is Nothing, "NULL", da.SqlEncode(Me.ForceWidthTo)) & ", ForceSortTo = " & If(Me.ForceSortTo Is Nothing, "NULL", da.SqlEncode(Me.ForceSortTo)) & ",ForceFilterTo=" & If(Me.ForceFilterTo Is Nothing, "NULL", da.SqlEncode(Me.ForceFilterTo)) & " , FK_DisplayUnit_ID = " & If(Me.DisplayUnit Is Nothing, "NULL", Me.DisplayUnit.ID) & " where Path = " & da.SqlEncode(Me.Path) & " AND [FK_Screen_Id]=" & Me.ScreenID & " and FK_Account_Id=" & Me.AccountID & " AND FK_Field_Id = " & Me.FieldId
        Return da.DBExecutesql(sql$, False) > 0
    End Function

    Public Function Insert() As Boolean
        Dim sql$
        sql$ = "INSERT INTO AccountScreenOverride (FK_Account_Id,FK_Screen_Id,Path,FK_Field_Id,ForceVisibilityTo,ForceOrderTo,ForceWidthTo,ForceSortTo,ForceFilterTo,FK_DisplayUnit_ID) VALUES (" & Me.AccountID & "," & Me.ScreenID & "," & If(Me.Path Is Nothing, "NULL", da.SqlEncode(Me.Path)) & "," & Me.FieldId & "," & If(Me.ForceVisibilityTo Is Nothing, "NULL", da.SqlEncode(Me.ForceVisibilityTo)) & "," & If(Me.ForceOrderTo Is Nothing, "NULL", da.SqlEncode(Me.ForceOrderTo)) & "," & If(Me.ForceWidthTo Is Nothing, "NULL", da.SqlEncode(Me.ForceWidthTo)) & "," & If(Me.ForceSortTo Is Nothing, "NULL", da.SqlEncode(Me.ForceSortTo)) & "," & If(Me.ForceFilterTo Is Nothing, "NULL", da.SqlEncode(Me.ForceFilterTo)) & "," & If(Me.DisplayUnit Is Nothing, "NULL", Me.DisplayUnit.ID) & ")"
        Return da.DBExecutesql(sql$, False) > 0
    End Function



End Class

File: IQ/Classes/clsScreenOverride.cs
using System.Runtime.Serialization;
using dataAccess;


[DataContract]
public class clsScreenOverride
{
    [DataMember]
    public int AccountID;
    [DataMember]
    public int ScreenID;
    [DataMember]
    public string Path;
    [DataMember]
    public int FieldId;
    [DataMember]
    public Nullable<bool> ForceVisibilityTo;
    [DataMember]
    public Nullable<int> ForceOrderTo;
    [DataMember]
    public Nullable<double> ForceWidthTo;
    [DataMember]
    public string ForceSortTo;
    [DataMember]
    public string ForceFilterTo;
    [DataMember]
    public string FieldName;
    [DataMember]
    public clsUnit DisplayUnit;


    public clsScreenOverride(int AccountID, int ScreenID, string BranchPath, int FieldId, Nullable<bool> ForceVisibilityTo, Nullable<int> ForceOrderTo, Nullable<double> ForceWidthTo, string ForceSortTo, string ForceFilterTo, clsUnit DisplayUnit)
    {
        this.AccountID = AccountID;
        this.FieldId = FieldId;
        this.ForceVisibilityTo = ForceVisibilityTo;
        this.ForceOrderTo = ForceOrderTo;
        this.ScreenID = ScreenID;
        this.Path = BranchPath;
        this.ForceWidthTo = ForceWidthTo;
        this.ForceSortTo = ForceSortTo;
        this.ForceFilterTo = ForceFilterTo;
        this.DisplayUnit = DisplayUnit;
        iq.ScreenOverrides.Add(this);
    }

    public clsScreenOverride()
    {

    }

    public bool Update()
    {

        object sql = null;
        sql = "Update AccountScreenOverride set ForceVisibilityTo = " + System.Convert.ToString(this.ForceVisibilityTo == null ? "NULL" : (da.SqlEncode(this.ForceVisibilityTo))) + ",ForceOrderTo = " + System.Convert.ToString(this.ForceOrderTo == null ? "NULL" : (da.SqlEncode(this.ForceOrderTo))) + ",ForceWidthTo = " + System.Convert.ToString(this.ForceWidthTo == null ? "NULL" : (da.SqlEncode(this.ForceWidthTo))) + ", ForceSortTo = " + System.Convert.ToString(this.ForceSortTo == null ? "NULL" : (da.SqlEncode(this.ForceSortTo))) + ",ForceFilterTo=" + System.Convert.ToString(this.ForceFilterTo == null ? "NULL" : (da.SqlEncode(this.ForceFilterTo))) + " , FK_DisplayUnit_ID = " + System.Convert.ToString(this.DisplayUnit == null ? "NULL" : this.DisplayUnit.ID) + " where Path = " + da.SqlEncode(this.Path) + " AND [FK_Screen_Id]=" + System.Convert.ToString(this.ScreenID) + " and FK_Account_Id=" + System.Convert.ToString(this.AccountID) + " AND FK_Field_Id = " + System.Convert.ToString(this.FieldId);
        return da.DBExecutesql(sql, false) > 0;
    }

    public bool Insert()
    {
        object sql = null;
        sql = "INSERT INTO AccountScreenOverride (FK_Account_Id,FK_Screen_Id,Path,FK_Field_Id,ForceVisibilityTo,ForceOrderTo,ForceWidthTo,ForceSortTo,ForceFilterTo,FK_DisplayUnit_ID) VALUES (" + System.Convert.ToString(this.AccountID) + "," + System.Convert.ToString(this.ScreenID) + "," + System.Convert.ToString(this.Path == null ? "NULL" : (da.SqlEncode(this.Path))) + "," + System.Convert.ToString(this.FieldId) + "," + System.Convert.ToString(this.ForceVisibilityTo == null ? "NULL" : (da.SqlEncode(this.ForceVisibilityTo))) + "," + System.Convert.ToString(this.ForceOrderTo == null ? "NULL" : (da.SqlEncode(this.ForceOrderTo))) + "," + System.Convert.ToString(this.ForceWidthTo == null ? "NULL" : (da.SqlEncode(this.ForceWidthTo))) + "," + System.Convert.ToString(this.ForceSortTo == null ? "NULL" : (da.SqlEncode(this.ForceSortTo))) + "," + System.Convert.ToString(this.ForceFilterTo == null ? "NULL" : (da.SqlEncode(this.ForceFilterTo))) + "," + System.Convert.ToString(this.DisplayUnit == null ? "NULL" : this.DisplayUnit.ID) + ")";
        return da.DBExecutesql(sql, false) > 0;
    }



}